hivemind/averaging/matchmaking.py

```diff
@@ -242,7 +240,7 @@ async def _request_join_group(self, leader: PeerID) -> Optional[GroupInfo]:
             logger.debug(f"{self} - potential leader {leader} did not respond within {self.request_timeout}")
             return None
         except (P2PHandlerError, ControlFailure, DispatchFailure, StopAsyncIteration) as e:
-            logger.debug(f"{self} - failed to request potential leader {leader}:")
+            logger.debug(f"{self} - failed to request potential leader {leader}: {repr(e)}", exc_info=True)
```

Comment:
```suggestion
            logger.debug(f"{self} - failed to request potential leader {leader}:", exc_info=True)
```

No need for the repr, the full traceback will be printed right away.